yarn-project/types/src/logs/function_l2_logs.ts

```diff
@@ -33,6 +35,28 @@ export class FunctionL2Logs {
     return this.logs.reduce((acc, log) => acc + log.length + 4, 0) + 4;
   }
 
+  /**
+   * Calculates hash of serialized logs.
+   * @returns 2 fields containing all 256 bits of information of sha256 hash.
+   */
+  public hash(): [Fr, Fr] {
+    const hash = sha256(this.toBuffer());
+
+    // TS version of https://github.com/AztecProtocol/aztec-packages/blob/e2e3bf1dbeda5199060fb1711200d20414557cd4/circuits/cpp/src/aztec3/circuits/hash.hpp#L330
+    // Split the hash into two fields, a high and a low
+    const buf1: Buffer = Buffer.alloc(32);
+    const buf2: Buffer = Buffer.alloc(32);
+
+    for (let i = 0; i < 16; i++) {
+      buf1[i] = 0;
```

Comment:
don't think you need to be doing this, as all bytes will be 0 when calling `Buffer.alloc(32)`

Reply:
in fact I think you could write this loop as 
```
buf1 = Buffer.concat([Buffer.alloc(16), hash.subarray(0, 16)])
buf2 = Buffer.cocnat([Buffer.alloc(16), hash.subarray(16, 16)])
```
Not sure if more performant though

Reply:
Good catch. I copied it from C++ where the values could be non-zero and hence needs to get set to 0. Will update.